pkg/cache/placeholder_manager.go

```diff
@@ -70,10 +70,28 @@ func (mgr *PlaceholderManager) createAppPlaceholders(app *Application) error {
 	return nil
 }
 
-// recycle all the placeholders for an application
-func (mgr *PlaceholderManager) Recycle(appID string) {
-	log.Logger().Info("start to recycle app placeholders",
-		zap.String("appID", appID))
+// clean up all the placeholders for an application
+func (mgr *PlaceholderManager) CleanUp(app *Application) {
+	log.Logger().Info("start to clean up app placeholders",
+		zap.String("appID", app.GetApplicationID()))
+	for taskID, task := range app.taskMap {
+		if task.GetTaskPlaceholder() {
+			// remove pod
+			err := mgr.clients.KubeClient.Delete(task.pod)
+			if err != nil {
+				log.Logger().Error("failed to clean up placeholder pod",
+					zap.Error(err))
+			}
+			// remove task from application
+			err = app.removeTask(taskID)
+			if err != nil {
+				log.Logger().Error("failed to remove task from application",
+					zap.Error(err))
+			}
```

Comment:
second thought, let's remove this. We do not need to remove the task from the app while cleaning up the placeholders. Task needs to be cleaned up separately, the normal process involves the core side to release corresponding resources in the scheduler.

Reply:
OK, done.